weblab/entities/views.py

```diff
@@ -1013,3 +1013,43 @@ def get(self, request, *args, **kwargs):
         return JsonResponse({
             task: result
         })
+
+
+class EntityRunExperimentView(LoginRequiredMixin, EntityTypeMixin, DetailView):
+    """
+    Class for listing the possible experiment combinations
+    """
+    context_object_name = 'entity'
+    template_name = 'entities/entity_runexperiments.html'
+
+    def get_context_data(self, **kwargs):
+        entity = self.object
+
+        context = super().get_context_data(**kwargs)
+
+        # preposition to use in sentence: You may run this entity on/under the following entities
+        context['preposition'] = 'under'
+        if entity.entity_type == 'protocol':
+            context['preposition'] = 'on'
+
+        # ended up using a nested dict as nessed lists caused django's unpacking in forloops to
+        # mess things up slightly
+        other_entities = Entity.objects.filter(entity_type=entity.other_type)
+        context['object_list'] = []
+        for item in other_entities:
+            versions = item.cachedentity.versions
+            version_info = []
+            for version in versions.prefetch_related('tags'):
+                tag_list = list(version.tags.values_list('tag', flat=True))
+                commit = item.repo.get_commit(version.sha)
```

Comment:
If real-world performance turns out to be an issue we can probably do some clever tricks with repocache here, but not worth worrying about for now.